lib/logstash/inputs/stdin.rb

```diff
@@ -31,16 +36,21 @@ def run(queue)
         end
       rescue IOError, EOFError, LogStash::ShutdownSignal
         # stdin closed or a requested shutdown
+        @stop_requested.make_true
         break
+      rescue => e
+        # ignore any exception in the shutdown process
+        break if @stop_requested.true?
+        raise(e)
```

Comment:
This block of code seems unnecessary to me.
```ruby
rescue => e
# ignore any exception in the shutdown process
break if @stop_requested.true?
raise(e)
```
If an non-rescued exception is throw it will break out of the while loop?